bin/reth-bench/src/bench/new_payload_only.rs

```diff
@@ -0,0 +1,194 @@
+        //
+        // the jwt is hex encoded so we will decode it after
+        let jwt = std::fs::read_to_string(auth_jwt)?;
+        let jwt = JwtSecret::from_hex(jwt)?;
+
+        // get engine url
+        let auth_url = Url::parse(&self.benchmark.engine_rpc_url)?;
+
+        // construct the authed transport
+        info!("Connecting to Engine RPC at {} for replay", auth_url);
+        let auth_transport = AuthenticatedTransportConnect::new(auth_url, jwt);
+        let client = ClientBuilder::default().connect_boxed(auth_transport).await?;
+        let auth_provider = RootProvider::<_, AnyNetwork>::new(client);
+
+        let first_block = match benchmark_mode {
+            BenchMode::Continuous => {
+                // fetch Latest block
+                block_provider.get_block_by_number(BlockNumberOrTag::Latest, true).await?.unwrap()
+            }
+            BenchMode::Range(ref mut range) => {
+                match range.next() {
+                    Some(block_number) => {
+                        // fetch first block in range
+                        block_provider
+                            .get_block_by_number(block_number.into(), true)
+                            .await?
+                            .unwrap()
+                    }
+                    None => {
+                        // return an error
+                        panic!("RangeEmpty");
+                    }
+                }
+            }
+        };
+
+        let mut next_block = match first_block.header.number {
+            Some(number) => {
+                // fetch next block
+                number + 1
+            }
+            None => {
+                // this should never happen
+                // TODO: log or return error, we should probably not return the
+                // block here
+                panic!("BlockNumberNone");
+            }
+        };
+
+        let (sender, mut receiver) = tokio::sync::mpsc::channel(1000);
+        tokio::task::spawn(async move {
+            while benchmark_mode.contains(next_block) {
+                let block_res = block_provider.get_block_by_number(next_block.into(), true).await;
+                let block = block_res.unwrap().unwrap();
+                let block = match block.header.hash {
+                    Some(block_hash) => {
+                        // we can reuse the hash in the response
+                        Block::try_from(block).unwrap().seal(block_hash)
+                    }
+                    None => {
+                        // we don't have the hash, so let's just hash it
+                        Block::try_from(block).unwrap().seal_slow()
+                    }
+                };
+
+                next_block += 1;
+                sender.send(block).await.unwrap();
+            }
+        });
+
+        // put results in a summary vec so they can be printed at the end
+        let mut results = Vec::new();
+        let total_benchmark_duration = Instant::now();
```

Comment:
looks like all of this is reused for the `BenchmarkCommand`, could you please move as much of it as possible to one or more functions

Reply:
made a `BenchContext` struct for the provider etc intialization